services/tradingService.js

```diff
@@ -44,35 +47,57 @@ const trade = async (tradeData) => {
 
     switch (tradeType) {
       case 'SELL': {
+        if (!userStocks.id || userStocks.quantity < quantity) {
+          return { canUserTrade: false, errorMessage: INSUFFICIENT_QUANTITIES }
+        }
+
         quantityToUpdate = quantity + stockData.quantity
-        userBalance = quantity * stockData.price
-        updatedCurrencyData.dinero = userBalance + currencies.dinero
-        stockPriceToBeUpdated = getUpdatedPrice(stockData.price)
+        userBalance = (quantity * stockData.price) + currencies.dinero
```

Comment:
Please use constants for currency names -- dinero, neelam. We already have a constants file

Reply:
a324cbc